internal/clients/aws.go

```diff
@@ -127,11 +144,22 @@ func getAWSConfigWithDefaultRegion(ctx context.Context, c client.Client, obj run
 		return nil, err
 	}
 	if cfg.Region == "" && obj.GetObjectKind().GroupVersionKind().Group == "iam.aws.upbound.io" {
-		cfg.Region = "us-east-1"
+		cfg.Region = getIAMRegion(pc)
 	}
 	return cfg, nil
 }
 
+func getIAMRegion(pc *v1beta1.ProviderConfig) string {
+	defaultRegion := "us-east-1"
+	if pc == nil || pc.Spec.Endpoint == nil || pc.Spec.Endpoint.PartitionID == nil {
+		return defaultRegion
```

Comment:
Please see the comment regarding the `iamRegions` map above. We should not be using `us-east-1` for the partitions other than `aws` and this implementation leaves the burden of configuring the right partition for an IAM resource to the user by specifying a proper `ProviderConfig`. While we should allow for overrides via the `ProviderConfig` (as it's already part of the `ProviderConfig` API, and for cases where the SDK has not updated its known partitions), we can provide a better UX with a robust & low maintenance cost implementation that relies on the `aws-go-sdk`.

Reply:
@ulucinar while I understand the comment about `iamRegions` map, I don't think it is directly relevant for the particular case of IAM resources. Note that this code part is only called for IAM MRs.

> this implementation leaves the burden of configuring the right partition for an IAM resource to the user by specifying a proper ProviderConfig

IIUC, unlike other resources, we anyways need some "partition-like" input at the `ProviderConfig` level as IAM resources are "regionless", so there is nothing else from which we can "infer" the partition, by only looking at the MR. Configuring a partition is actually a requirement rather than a config burden.

Alternative can be introducing an optional "partition" input for IAM MR specs, but I don't think this has a big benefit. Because the associated credentials of the provider config are "partition-bound" as described [here](https://docs.aws.amazon.com/whitepapers/latest/aws-fault-isolation-boundaries/partitions.html), i.e. they will be only valid for a single particular partition. The credentials & ProviderConfig actually "dictates" the partition already. 

Unlike handling of regions, where it is possible to use the same provider config and switch regions at MR, this does not make sense for partitions.